com.unity.netcode.gameobjects/Runtime/Connection/NetworkClient.cs

```diff
@@ -7,30 +7,95 @@ namespace Unity.Netcode
     /// </summary>
     public class NetworkClient
     {
+        /// <summary>
+        /// Returns true if the session instance is considered a server
+        /// </summary>
+        public bool IsServer { get; internal set; }
+
+        /// <summary>
+        /// Returns true if the session instance is considered a client
+        /// </summary>
+        public bool IsClient { get; internal set; }
+
+        /// <summary>
+        /// Returns true if the session instance is considered a host
+        /// </summary>
+        public bool IsHost => IsClient && IsServer;
+
+        /// <summary>
+        /// When true, the client is connected, approved, and synchronized with
+        /// the server.
+        /// </summary>
+        public bool IsConnected { get; internal set; }
+
+        /// <summary>
+        /// Is true when the client has been approved.
+        /// </summary>
+        public bool IsApproved { get; internal set; }
```

Comment:
Even though I agree in principle that having these properties here make more sense, instead of NetworkManager, it adds more onto our public API surface/footprint.

I will revert this public API change in NetworkClient and use references to NetworkManager's.

We can (and will) revisit this when we get to breaking API changes later in the future.

Reply:
I saw that, I'll give it a try and pick between two.